src/domain/fiat/display-currency.ts

```diff
@@ -68,3 +68,40 @@ export const getCurrencyMajorExponent = (
     return MajorExponent.STANDARD
   }
 }
+
+export const newDisplayAmountFromNumber = <T extends DisplayCurrency>({
+  amount,
+  currency,
+}: {
+  amount: number
+  currency: T
+}): NewDisplayAmount<T> | ValidationError => {
+  const amountInMinor = safeBigInt(amount)
+  if (amountInMinor instanceof Error) return amountInMinor
+
+  const displayMajorExponent = getCurrencyMajorExponent(currency)
+
+  return {
+    amountInMinor,
+    currency,
+    displayInMajor: (Number(amountInMinor) / 10 ** displayMajorExponent).toFixed(
+      displayMajorExponent,
+    ),
+  }
+}
+
+export const priceAmountFromNumber = <T extends DisplayCurrency>({
```

Comment:
Implementing because we've swapped in `NewDisplayAmount` now and we use bigints everywhere for display amounts, but this price value still needs to be a `number` for now. That's why I made the new `PriceAmount` type and this complementary method.